File: Python/8kyu/8kyu-multiply.py
# Multiply
# Difficulty: 8 kyu

# Task: fix the code
def multiply(a, b):
    a * b

# Solution
def multiply(a, b):
    return a * b

# Tests

import codewars_test as test
from solution import multiply

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(multiply(2,1), 2)

File: R/8kyu/8kyu-multiply.r
# Multiply
# Difficulty: 8 kyu

# Task
mul <- function(a, b) {
  a * c # try to figure out why it doesn't work!
}

# Solution
mul <- function(a, b) {
  a * b
}


# Tests

test_that("test for associativity of multiplication", {
  a <- runif(1, 0, 10000)
  b <- runif(1, 0, 10000)
  c <- runif(1, 0, 10000)
  expect_equal(mul(a, b), a * b)
  expect_equal(mul(mul(a, b), c), a * b * c)
  expect_equal(mul(a, mul(b, c)), a * b * c)
})